File: python_helpers/results.py
from flask import Blueprint, jsonify
import json

results_bp = Blueprint('results', __name__, url_prefix='/server-api')

@results_bp.route('/results')
def get_results():
    with open('data/results.json') as f:
        results = json.load(f)
    return jsonify(results)

File: utils/results.js
/**
 * Fetches results from the data source.
 * @returns {Promise<Object>} - The results data.
 */
export async function getResults() {
  // In a real Next.js app, this might read from a file or database.
  // For now, mirroring the Flask route's intention.
  try {
    const response = await fetch(`${process.env.API_URI || ''}/data/results.json`);
    return await response.json();
  } catch (error) {
    console.error("Error fetching results:", error);
    return null;
  }
}
